app/javascript/javascript/filtersUtils.js

```diff
@@ -74,21 +74,44 @@ export const setupTimeRangeFilter = (
     );
   }
 };
-export const setupAutocomplete = (callback, id, path, createParams) => {
-  if (document.getElementById(id)) {
-    $(`#${id}`).autocomplete({
+export const setupTagsAutocomplete = (callback, path, createParams) => {
+  if (document.getElementById("tags")) {
+    $(`#tags`)
+      .autocomplete({
+        source: function(request, response) {
+          const data = {
+            q: { input: request.term, ...createParams() }
+          };
+          $.getJSON(path, data, response);
+        },
+        select: function(event, ui) {
+          callback(ui.item.value);
+        },
+        minLength: 0
+      })
+      .focus(function() {
+        $(this).autocomplete("search");
+      });
+  } else {
+    window.setTimeout(setupAutocomplete(callback, path), 100);
+  }
+};
+
+export const setupProfileNamesAutocomplete = callback => {
+  if (document.getElementById("profile-names")) {
+    $(`#profile-names`).autocomplete({
       source: function(request, response) {
         const data = {
-          q: { input: request.term, ...createParams() }
+          q: { input: request.term }
         };
-        $.getJSON(path, data, response);
+        $.getJSON("api/autocomplete/usernames", data, response);
```

Comment:
would be great to switch to fetch (whatwg-fetch) which we're already using in other places, e.g. `elm.js`
https://github.com/github/fetch

this is a cool source when you suspect you don't need jquery somewhere: http://youmightnotneedjquery.com/

Reply:
Again, not sure if it's possible with this jquery library. But we can try get rid of it totally :)